Mathlib/Algebra/Order/Monoid/Canonical/Basic.lean

```diff
@@ -0,0 +1,18 @@
+/-
+Copyright (c) 2024 Yaël Dillies. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Yaël Dillies
+-/
+import Mathlib.Algebra.Order.Monoid.Canonical.Defs
+import Mathlib.Data.Finset.Lattice
+
+/-!
+# Extra lemmas about canonically ordered monoids
+-/
+
+namespace Finset
+variable {ι α : Type*} [CanonicallyLinearOrderedAddCommMonoid α] {s : Finset ι} {f : ι → α}
+
+@[simp] lemma sup_eq_zero : s.sup f = 0 ↔ ∀ i ∈ s, f i = 0 := by simp [← bot_eq_zero']
```

Comment:
Should we have `inf_eq_zero` with `Exists` as well?

Reply:
We don't have the analogous lemma for `bot` at the moment